WebKitBrowser/WebKitImplementation.cpp

```diff
@@ -1911,6 +1931,144 @@ static GSourceFuncs _handlerIntervention =
+            return true;
+        }
+
+        bool parseRssFromStatmLine(const std::string &statmLine, uint32_t &inBytes)
+        {
+            std::vector<std::string> items = splitString(statmLine, ' ');
+            if (items.size() < 7)
+            {
+                SYSLOG(Trace::Error, (_T("Unexpected size(%u) of 'statm' line."), items.size()));
+                return false;
+            }
+            static const long PageSize = sysconf(_SC_PAGE_SIZE);
+            unsigned long rssPageNum = std::stoul(items[1]);
+            inBytes = rssPageNum * PageSize;
+            return true;
+        }
+
+        void collectMetricsOnLoadStart()
+        {
+            if (_didSendLaunchMetrics)
+                 return;
+
+            auto getProcessLaunchStateString = [&]() -> std::string
+            {
+                switch(_webProcessState)
+                {
+                    case WebProcessCold: return "Cold";
+                    case WebProcessHot:  return "Hot";
+                }
+                return "Unknown";
+            };
+
+            auto addSystemInfo = [&](std::map<std::string, std::string> &metrics)
+            {
+                struct sysinfo info;
+                if (sysinfo(&info) != 0)
+                {
+                    SYSLOG(Trace::Error, (_T("Failed to get sysinfo error=%d."), errno));
+                    return;
+                }
+                static const long NPROC_ONLN = sysconf(_SC_NPROCESSORS_ONLN);
+                static const float LA_SCALE = static_cast<float>(1 << SI_LOAD_SHIFT);
+                metrics["MemTotal"] = std::to_string(info.totalram * info.mem_unit);
+                metrics["MemFree"] = std::to_string(info.freeram * info.mem_unit);
+                metrics["MemSwapped"] = std::to_string((info.totalswap - info.freeswap) * info.mem_unit);
+                metrics["Uptime"] = std::to_string(info.uptime);
+                metrics["LoadAvg"] = std::to_string(info.loads[0] / LA_SCALE) + " " +
+                                 std::to_string(info.loads[1] / LA_SCALE) + " " +
+                                 std::to_string(info.loads[2] / LA_SCALE);
+                metrics["NProc"] = std::to_string(NPROC_ONLN);
+            };
+            auto addProcessInfo = [&](std::map<std::string, std::string> &metrics)
+            {
+
+                pid_t webprocessPID = GetWebGetProcessIdentifier();
+                uint32_t rssInBytes = 0;
+                std::string statmLine;
+                if (readStatmLine(webprocessPID, statmLine))
+                {
+                    parseRssFromStatmLine(statmLine, rssInBytes);
+                }
+
+                metrics["ProcessRSS"] = std::to_string(rssInBytes);
+                metrics["ProcessPID"] = std::to_string(webprocessPID);
+                metrics["AppName"] = _URL.c_str();
+                metrics["webProcessStatmLine"] = statmLine;
+            };
+
+            std::map<std::string, std::string> metrics;
```

Comment:
Let's hold metrics in JsonObject, so you can convert them into JSON string.  to make it simple write directly to _launchMetricsMetrics. i.e.:

_launchMetricsMetrics.Reset();
_launchMetricsMetrics.Set(..., ...)